spring-amqp/src/main/java/org/springframework/amqp/support/converter/AbstractJackson2MessageConverter.java

```diff
@@ -205,7 +223,15 @@ public Object fromMessage(Message message, @Nullable Object conversionHint) thro
 					encoding = getDefaultCharset();
 				}
 				try {
-					if (conversionHint instanceof ParameterizedTypeReference) {
+					JavaType inferredType = this.javaTypeMapper.getInferredType(properties);
+					if (inferredType != null && this.useProjectionForInterfaces && inferredType.isInterface()
+							&& !inferredType.getRawClass().getPackage().getName().startsWith("java.util")) { // List etc
+						if (this.projectingConverter == null) {
+							this.projectingConverter = new ProjectingMessageConverter(this.objectMapper);
```

Comment:
Don't we need to reject such a configuration if there is no `spring-data-common` in the classpath?

Well, I believe we can do that check in the `setUseProjectionForInterfaces()` proactively.

Reply:
Can't we create this directly in the `setUseProjectionForInterfaces()` ?